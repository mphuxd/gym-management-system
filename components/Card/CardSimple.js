import React from 'react';
import { ArrowRight } from '@carbon/icons-react';

export const CARD_SIMPLE_TEST_ID = 'cardSimpleTestId';

function CardSimple({ heading, description }) {
  return (
    <div
      className="flex flex-col justify-between h-52 p-4 bg-white focus:bg-slate5 active:bg-slate5 shadow-xl shadow-mauve7 hover:shadow-mauve8 "
      data-testid={CARD_SIMPLE_TEST_ID}
    >
      <div className="flex flex-col gap-y-3">
        <h2 className="text-2xl font-medium">{heading}</h2>
        <p className="max-w-sm text-lg">{description}</p>
      </div>
      <ArrowRight className="text-red11" size={24} />
    </div>
  );
}

export default CardSimple;
