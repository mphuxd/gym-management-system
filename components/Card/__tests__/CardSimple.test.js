import React from 'react';
import { render, screen } from '@testing-library/react';
import CardSimple, { CARD_SIMPLE_TEST_ID } from '../CardSimple';

describe('CardSimple ', () => {
  it('should match snapshot', async () => {
    const heading = 'Card Heading';
    const description = 'Card Description';
    render(<CardSimple heading={heading} description={description} />);

    const cardSimple = await screen.findByTestId(CARD_SIMPLE_TEST_ID);
    expect(cardSimple).toHaveTextContent(heading);
    expect(cardSimple).toHaveTextContent(description);
    expect(cardSimple).toMatchSnapshot();
  });
});
