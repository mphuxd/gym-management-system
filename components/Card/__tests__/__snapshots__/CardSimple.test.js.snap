// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSimple  should match snapshot 1`] = `
<div
  class="flex flex-col justify-between h-52 p-4 bg-white focus:bg-slate5 active:bg-slate5 shadow-xl shadow-mauve7 hover:shadow-mauve8 "
  data-testid="cardSimpleTestId"
>
  <div
    class="flex flex-col gap-y-3"
  >
    <h2
      class="text-2xl font-medium"
    >
      Card Heading
    </h2>
    <p
      class="max-w-sm text-lg"
    >
      Card Description
    </p>
  </div>
  <svg
    aria-hidden="true"
    class="text-red11"
    fill="currentColor"
    focusable="false"
    height="24"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14 4L12.9 5.1 18.9 11.2 2 11.2 2 12.8 18.9 12.8 12.9 18.9 14 20 22 12z"
    />
  </svg>
</div>
`;
